### PR TITLE
Update Vendor String in python files to ssgcommon.py

### PR DESCRIPTION
**Pull request by redhatrises**

### CODE REVIEW

**Comment by redhatrises** [comment]
Would like this in 0.1.32, but understand if it should be pushed off to 0.1.33.

**Comment by redhatrises** [comment]
@shawndwells thoughts on this PR?

**Comment by shawndwells** [comment]
Love it. Like the OSCAP_VENDOR idea.

I stay away from ack'ing PRs into the core OpenSCAP code - don't want @mpreisler to come after me :)

**Comment by yuumasato** [comment]
I think it is interesting, but I'll leave it to Martin to decide.
But I'll do the release soon and I think he won't see it in time.

**Comment by shawndwells** [comment]
reminder, because I know I'll forget: If/when this gets merged, we'll need to take a look at peripheral documentation on open-scap.org, man pages, etc.

**Comment by jan-cerny** [comment]
This is nice, but I think if we change the profile and rule IDs, it will break existing tailoring files, so the users will have to create new tailoring files or edit them.

**Comment by mpreisler** [comment]
> This is nice, but I think if we change the profile and rule IDs, it will break existing tailoring files, so the users will have to create new tailoring files or edit them.

+1

Making IDs look better is "nice" but if it breaks user tailoring files and user command line snippets it's NACK IMO.

**Comment by shawndwells** [comment]
IMHO I want this now, there's never a good time for such changes. But if we change mid-RHEL7 it will cause problems with downstream users.

So - good idea, but wait until RHEL8 as to not interrupt users?

**Comment by mpreisler** [comment]
I am split on this. Advantages of this change have already been listed here, let me list some disadvantages.

Right now we ship SSG on various distributions, including RHEL, Fedora, Debian and others. They all have more or less the same IDs. That means that tutorials for remote evaluating Debian will work on all the distros where we ship. If we start doing vendor specific IDs we will break that.

SCAP Workbench on OSX and Windows will probably continue to use the vendor neutral IDs. Which means it's "dry run" command-line will not work on RHEL. From what I have heard that is a fairly popular feature.

tl;dr: I see a lot of value in having the same profile IDs everywhere.

**Comment by redhatrises** [comment]
At the very least, org.ssgproject is wrong because ssgproject.org does not exist. So, either ssgproject.org should redirect to open-scap.org, or the IDs should change to org.open-scap

From a non-technical/non-maintenance side, if Red Hat, for example, ships SSG with org.open-scap instead of com.redhat, that is not only confusing, but it is like saying that SSG is the content author for Red Hat instead of Red Hat being the content author. Same argument for Debian, Fedora, etc. 

From NIST, "namespace is a reverse-DNS style string associated with the content author."

**Comment by shawndwells** [comment]
We can register ssgproject.org without issue, but would rather change it to an existing domain (open-scap.org, openscap.io, whatever). Agree with @mpreisler this would be an invasive change... but there really is no good time to make it.

Part of the challenge is how to handle this for downstreams, who (right or wrong) use the same namespace. If we change now, RHEL7 users would have a namespace change in the middle of the RHEL7 lifecycle. Would put the burden on RHEL Release Engineering guys to change from upstream name back to org.ssgproject (high level of annoyance).

At this point, while IMHO is the "wrong answer," I don't see any path forward other than registering org.ssgproject for now and changing things for RHEL8.

**Comment by mpreisler** [comment]
> At this point, while IMHO is the "wrong answer," I don't see any path forward other than registering org.ssgproject for now and changing things for RHEL8.

+1

I don't like the solution but it is the best we have right now. Probably better than breaking every tutorial we have :-)

**Comment by shawndwells** [comment]
On 4/6/17 9:32 AM, Martin Preisler wrote:
>
>     At this point, while IMHO is the "wrong answer," I don't see any
>     path forward other than registering org.ssgproject for now and
>     changing things for RHEL8.
>
> +1
>
> I don't like the solution but it is the best we have right now.
> Probably better than breaking every tutorial we have :-)
>

Blocked off time tomorrow (Fri) to register the domain. Other ones like
openscap.io need renewal, so will take care of them all together. I'll
send a note to the public list once done.

**Comment by mpreisler** [comment]
At least for now this is a NACK, closing. Please reopen if you disagree or if there is any more input here.

**Comment by redhatrises** [comment]
We need to change upstream to be `org.open-scap`. Internally, downstream should be changed to `com.redhat`. These changes are being planned for RHEL8

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ set(SSG_MINOR_VERSION 1)
 set(SSG_PATCH_VERSION 31)
 set(SSG_VERSION "${SSG_MAJOR_VERSION}.${SSG_MINOR_VERSION}.${SSG_PATCH_VERSION}")
 
-set(SSG_VENDOR "ssgproject" CACHE STRING "Specify the XCCDF 1.2 vendor string.")
+set(SSG_VENDOR "open-scap" CACHE STRING "Specify the XCCDF 1.2 vendor string.")
```
Comment (mpreisler) [inline CMakeLists.txt]: org should be part of the vendor string IMO, it's not part of the XCCDF ID pattern, without it being in the pattern we are unable to say that com.redhat is the vendor. at best we can say that org.redhat is the vendor.

**Comment by mpreisler** [comment]
@redhatrises ok, rebase please. We also need to include "org." in the vendor string, else it's impossible to set `com.redhat` as the vendor.

**Comment by redhatrises** [comment]
@mpreisler made code updates. Will have a separate PR to update the vendor string.

File: shared/transforms/pcidss/transform_benchmark_to_pcidss.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
 def construct_xccdf_group(id_, desc, children, rules, rule_usage_map):
     ret = ElementTree.Element("{%s}Group" % (XCCDF_NAMESPACE))
-    ret.set("id", "xccdf_org.ssgproject.content_group_pcidss-req-%s" % (id_))
+    ret.set("id", ssgcommon.OSCAP_GROUP_PCIDSS + "-%s" % (id_))
```
Comment (mpreisler) [inline shared/transforms/pcidss/transform_benchmark_to_pcidss.py]: must resist to comment this syntactical awkwardness :D

**Comment by mpreisler** [comment]
ACK

**Comment by mpreisler** [comment]
> Would like this in 0.1.32, but understand if it should be pushed off to 0.1.33.

lol
